types/react/test/managedAttributes.tsx

```diff
@@ -182,27 +115,28 @@ interface LeaveMeAloneDtslint { foo: string; }
 // const memoTests = [
 //     // $ExpectError
 //     <MemoFunctionalComponent />,
-//     // $ExpectError won't work as long as FunctionalComponent doesn't work either
+//     // Requires TypeScript >= 3.2; see comment re FunctionalComponent above
 //     <MemoFunctionalComponent str='abc' />,
 //     // $ExpectError
 //     <MemoAnnotatedDefaultProps />,
 //     <AnnotatedDefaultProps str='abc' />,
-//     // $ExpectError this doesn't work despite JSX.LibraryManagedAttributes returning the correct type
+//     // Requires TypeScript >= 3.2; see comment re FunctionalComponent above
 //     <MemoAnnotatedDefaultProps str='abc' />,
-//     // $ExpectError won't work as long as FunctionalComponent doesn't work either
+//     // Requires TypeScript >= 3.2; see comment re FunctionalComponent above
 //     <LazyMemoFunctionalComponent str='abc' />,
 //     // $ExpectError
 //     <LazyMemoAnnotatedDefaultProps />,
-//     // $ExpectError this doesn't work despite JSX.LibraryManagedAttributes returning the correct type
+//     // Requires TypeScript >= 3.2; see comment re FunctionalComponent above
 //     <LazyMemoAnnotatedDefaultProps str='abc' />
 // ];
 
+// // $ExpectType Pick<Props, "bool" | "node" | "num" | "str"> & Partial<Pick<Props, "fnc" | "reqNode">>
 // type AnnotatedDefaultPropsLibraryManagedAttributes = JSX.LibraryManagedAttributes<typeof AnnotatedDefaultProps, Props>;
-// // $ExpectType AnnotatedDefaultPropsLibraryManagedAttributes
+// // $ExpectType Pick<Props, "bool" | "node" | "num" | "str"> & Partial<Pick<Props, "fnc" | "reqNode">>
```

Comment:
The alias doesn't work? 😞

Reply:
dtslint does a [string comparison](https://github.com/Microsoft/dtslint/blob/73a196741d5b6ca9af7f6ef68ef4ee3e801ac4b5/src/rules/expectRule.ts#L302) of the expected type to the default string representation of the actual type, so I don't think this could ever have worked.  Feel free to file an enhancement request against dtslint.  In the meantime, if we really wanted to assert the equality of two types without writing out one of them in full, there are probably other ways, but in this case it didn't seem too bad.